projects/plugins/crm/src/js/components/automations-admin/components/workflows-home/index.tsx

```diff
@@ -0,0 +1,37 @@
+import { useCallback } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { BulkWorkflowActions } from '../bulk-workflow-actions';
+import { EditModal } from '../edit-modal';
+import { WorkflowTable } from '../workflow-table';
+import styles from './styles.module.scss';
+import type { AutomationsState } from 'crm/state/automations-admin/reducer';
+import type { Workflow } from 'crm/state/automations-admin/types';
+
+type WorkflowsHomeProps = {
+	workflows: AutomationsState[ 'workflows' ];
+};
+
+export const WorkflowsHome: React.FC< WorkflowsHomeProps > = ( { workflows } ) => {
+	const { id } = useParams< { id: string } >();
+
+	let workflow: Workflow | undefined;
+	if ( id ) {
+		const workflowId = Number( id );
+		workflow = workflows[ workflowId ];
+	}
+
+	const navigate = useNavigate();
+	const onEditModalClose = useCallback( () => {
+		navigate( '/automations/manage', { replace: true } );
+	}, [ navigate ] );
+
+	return (
+		<div className={ styles.container }>
+			<BulkWorkflowActions />
+			<WorkflowTable workflows={ Object.values( workflows ) } />
+			{ workflow && (
+				<EditModal isOpen={ !! workflow } onClose={ onEditModalClose } workflow={ workflow } />
+			) }
+		</div>
+	);
```

Comment:
I was wondering if it might be worth wrapping any of the components that might be more likely to have errors or be broken, in an error boundary? Only if that's something that could happen on occasion.

Something like:
```
<ErrorBoundary fallback={<p>Something holding text</p>}>
  <WorkflowTable workflows={ Object.values( workflows ) } />
</ErrorBoundary>
```
or whatever is more likely to cause issues? Along with creating the error boundary class etc. 

Maybe that's more an enhancement idea for the future (if it's relevant) since we're not making any of this visible just now?

Reply:
We should implement something like this but I considered it out of scope for this PR. The REST API calls could have errors (e.g. 4XX responses), which I would usually handle with something like
```
const { data, status, error } = useQuery( ... );

if ( error ) {
    return <ErrorDisplay error={ error }/>;
}

if ( status === 'loading' ) {
   return <LoadingIndicator />;
}

return <DataDisplay data={ data }/>
```

There are some generic catch all error handlers in Calypso and I think in the Jetpack plugin, but they're a part of an overall notification system which displays messages in popovers and we don't have that in CRM yet (at least not in the React part). The Figma design does include these so we will need to implement them at some point:

<img width="940" alt="image" src="https://github.com/Automattic/jetpack/assets/42627630/67598401-a021-45a3-a9af-a88b28de4c1c">

Reply:
Just a heads up about the notifications that we have RNA components for it which hopefully makes it a bit smaller task 👏